### PR TITLE
ANDROID-11221 Update plugin to upload to mavencentral

### PR DESCRIPTION
**Pull request by pmartinbTEF**
[ANDROID-11221](https://jira.tid.es/browse/ANDROID-11221)

Update plugin to update to mavencentral based on this PR: https://github.com/Telefonica/mistica-android/pull/188
and this migration guide: https://github.com/gradle-nexus/publish-plugin/wiki/Migration-from-gradle_nexus_staging-plugin---nexus_publish-plugin-duo

### CODE REVIEW

File: .github/workflows/publish.yml
Comment (pmartinbTEF) [inline .github/workflows/publish.yml:1]: For uniformity with the rest of the libraries, the publication of a Release must follow this procedure:  https://confluence.tid.es/pages/viewpage.action?spaceKey=CTO&title=Releasing+a+new+version+of+a+library

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -7,10 +7,13 @@ jobs:
...
-      uses: actions/checkout@v2
-    
-    - name: "Lint & Detekt"
-      uses: vgaidarji/android-github-actions-build@v1.0.1
-      with:
-        args: 'bash ./gradlew detekt check'
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Run checks
```
Comment (pmartinbTEF) [inline .github/workflows/tests.yml:13]: Different steps as in other libraries

File: app/src/main/java/com/telefonica/androidlogger/app/MainActivity.kt [hunk 2]
```diff
@@ -20,7 +20,7 @@ class MainActivity : AppCompatActivity() {
 
     private fun setupViews() {
         findViewById<Button>(R.id.button_add_logs).setOnClickListener {
-            for (i in 1..NUMBER_OF_LOGS_TO_ADD) {
```
Comment (pmartinbTEF) [inline app/src/main/java/com/telefonica/androidlogger/app/MainActivity.kt:23]: Unused i causes detekt error

**Review by dpastor — APPROVED** [review]
Thanks!! :)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -7,10 +7,13 @@ jobs:
...
+        if: success() || failure()
+        run: ./gradlew check
+
+      - name: Run Detekt
+        if: success() || failure()
+        run: ./gradlew detekt
```
Comment (gmerinojimenez) [inline .github/workflows/tests.yml:19]: Doesn't check execute also detekt?

**Review by gmerinojimenez — APPROVED** [review]
Thanks!!

File: .github/workflows/tests.yml (hunk 3 above)
Comment (pmartinbTEF) [reply .github/workflows/tests.yml:19]: According to doc https://detekt.dev/docs/gettingstarted/gradle/ it should be run with check task but I tested adding a detekt error and it is not be catched with ./gradlew check

**Pull request merged**